exec/java-exec/src/main/java/org/apache/drill/exec/expr/ClassGenerator.java

```diff
@@ -215,22 +219,47 @@ public JVar declareVectorValueSetupAndMember(DirectExpression batchName, TypedFi
     return vv;
   }
 
+  public enum RotateMode {
```

Comment:
The name 'RotateMode' seems a misnomer since there is no actual re-use or rotation of the JBlocks, right ?  It is confusing. Should we call this BlockCreationMode ?